### PR TITLE
Added support for data disk and disks' size

### PR DESCRIPTION
**Pull request by eldios**
## Description
Added support for different sizes for root and data disk in Terraform script.

## Type of change

* New feature (non-breaking change which adds functionality)

## Additional information

This feature is particularly useful when testing old device-mapper based setup where an additional data disk is needed.

### CODE REVIEW

File: terraform-dev/cluster.tf [hunk 1]
```diff
@@ -3,7 +3,12 @@ variable "image_name" {
   default = "ubuntu-18.04-server-cloudimg-amd64.img"
 }
 
-variable "disk_size" {
+variable "root_disk_size" {
+  type = string
```
Comment (a-palchikov) [inline terraform-dev/cluster.tf:7]: Can you add descriptions what the disks will be used for?

File: terraform-dev/cluster.tf [hunk 2]
```diff
@@ -51,7 +55,14 @@ resource "libvirt_volume" "root" {
...
   base_volume_id = element(libvirt_volume.os-qcow2.*.id, count.index)
   pool = "default"
-  size = var.disk_size
+  size = var.root_disk_size
+  count = var.nodes_count
+}
+
+resource "libvirt_volume" "data" {
+  name = "data-disk-${count.index}.qcow2"
+  pool = "default"
```
Comment (a-palchikov) [inline terraform-dev/cluster.tf:64]: nit: maybe add an option to override the pool with a default to `default`?

**Review by a-palchikov — APPROVED** [review]

**Comment by UnderGreen** [comment]
Why do we need a disk for the data directory?

File: terraform-dev/Makefile [hunk 3]
```diff
@@ -1,6 +1,7 @@
 IMAGEURL := https://cloud-images.ubuntu.com/releases/bionic/release/ubuntu-18.04-server-cloudimg-amd64.img
 CPU_COUNT ?= 1
-DISK_SIZE ?= 15000000000
+ROOT_DISK_SIZE ?= 15000000000
+DATA_DISK_SIZE ?= 15000000000
```
Comment (knisbet) [inline terraform-dev/Makefile:4]: Do both disks need to be this size (doubling the amount of disk space used?)

Comment (eldios) [reply terraform-dev/Makefile:4]: It won't use any additional disk space, unless needed. I actually set them both to be 50G just to be sure in the example below for 4 nodes:
```
193K /var/lib/libvirt/images/data-disk-0.qcow2
193K /var/lib/libvirt/images/data-disk-1.qcow2
193K /var/lib/libvirt/images/data-disk-2.qcow2
193K /var/lib/libvirt/images/data-disk-3.qcow2
899M /var/lib/libvirt/images/os-disk-0.qcow2
899M /var/lib/libvirt/images/os-disk-1.qcow2
899M /var/lib/libvirt/images/os-disk-2.qcow2
899M /var/lib/libvirt/images/os-disk-3.qcow2
12G /var/lib/libvirt/images/root-disk-0.qcow2
21G /var/lib/libvirt/images/root-disk-1.qcow2
20G /var/lib/libvirt/images/root-disk-2.qcow2
14G /var/lib/libvirt/images/root-disk-3.qcow2
```

File: terraform-dev/cluster.tf (hunk 2 above)
Comment (eldios) [reply terraform-dev/cluster.tf:64]: good point

File: terraform-dev/cluster.tf (hunk 1 above)
Comment (eldios) [reply terraform-dev/cluster.tf:7]: Done.

**Comment by eldios** [comment]
@UnderGreen it's called "data" just to differentiate it from the "root" one... I used it in my tests about the devicemapper storage but you could use it for anything really... it's just meant to be a "secondary disk".

**Review by r0mant — APPROVED** [review]

**Pull request merged**